src/test/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPosterTest.java

```diff
@@ -0,0 +1,114 @@
+package com.atlassian.bitbucket.jenkins.internal.status;
+
+import com.atlassian.bitbucket.jenkins.internal.client.*;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.TaskListener;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.PrintStream;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BuildStatusPosterTest {
+
+    private static final String REVISION_SHA1 = "67d71c2133aab0e070fb8100e3e71220332c5af1";
+    private static final String SERVER_ID = "FakeServerId";
+    private static final String SERVER_URL = "http://www.example.com";
+    @ClassRule
+    public static JenkinsRule jenkins = new JenkinsRule();
+    @Mock
+    private BitbucketRevisionAction action;
+    @Mock
+    private AbstractBuild build;
+    @InjectMocks
+    private BuildStatusPoster buildStatusPoster;
+    @Captor
+    private ArgumentCaptor<BitbucketBuildStatus> captor;
+    @Mock
+    private BitbucketClientFactory factory;
+    @Mock
+    private BitbucketClientFactoryProvider factoryProvider;
+    @Mock
+    private TaskListener listener;
+    @Mock
+    private PrintStream logger;
+    @Mock
+    private BitbucketPluginConfiguration pluginConfiguration;
+    @Mock
+    private BitbucketBuildStatusClient postClient;
+    @Mock
+    private AbstractProject project;
+    @Mock
+    private BitbucketServerConfiguration server;
+
+    @Before
+    public void setup() {
+        when(build.isBuilding()).thenReturn(true);
+        when(build.getId()).thenReturn("10");
+        when(build.getDurationString()).thenReturn("23 sec");
```

Comment:
this seems weird? is this how Jenkins does it? or is the `DurationString` only set when the build is complete?

Reply:
Jenkins has a `Messages.Run_InProgressDuration` method for when the build is in progress (although we do not display because as you say it's weird), but yes this is the format used. If we dislike the format that is output we can roll our own, build also provides a `getDuration`- if we prefer "seconds" to "secs" etc.

Reply:
na, all good. I'm more surprised that 'getDuration' returns something before the build is done. If it does, then all is fine.

Reply:
Herp derp. This doesn't need to return anything. I'll fix it.